tests/src/collect.rs

```diff
@@ -357,34 +579,83 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_attrs(&mut self) -> Attrs {
-        let mut parsed = AttrFlags::empty();
+        let mut stages = TestStages::empty();
+        let mut flags = AttrFlags::empty();
+        let mut pdf_standard = None;
         while !self.s.eat_if("---") {
-            let attr = self.s.eat_until(char::is_whitespace);
-            let flag = match attr {
-                "large" => AttrFlags::LARGE,
-                "html" => AttrFlags::HTML,
-                "render" => AttrFlags::RENDER,
-                "pdftags" => AttrFlags::PDFTAGS,
-                "nopdfua" => AttrFlags::NOPDFUA,
+            let attr_name = self.s.eat_while(is_id_continue);
+            let mut attr_params = None;
+            if self.s.eat_if('(') {
+                attr_params = Some(self.s.eat_until(')'));
+                if !self.s.eat_if(')') {
+                    self.error("expected closing parenthesis");
+                }
+            }
+            if !self.s.at(' ') {
+                self.error("expected a space after an attribute");
+            }
+
+            match attr_name {
+                "paged" => {
+                    // The paged target will test render, pdf, and svg by
+                    // default.
+                    self.set_attr(attr_name, &mut stages, TestStages::PAGED);
+                    stages |= TestStages::RENDER | TestStages::PDF | TestStages::SVG;
```

Comment:
this is a bit duplicate with the code in `CliArguments::stages`. And also a bit with `TestStages::paged_stages`, but that one is a bit different (which is a bit confusing to me).

overall, it might be cleaner if the parsing code is not concerned with how different targets imply each other

Reply:
I've refactored this now a little bit, let me know if this seems fine.